Update the guidance on Default Azure Credential usage (#267)
microsoft/agent-skills: .github/plugins/azure-sdk-typescript/skills/azure-storage-queue-ts/SKILL.md

```bash
AZURE_STORAGE_ACCOUNT_NAME=<account-name>
AZURE_STORAGE_ACCOUNT_KEY=<account-key>
```

## Authentication

> ### Entra ID TokenCredential (Recommended)

Reviewer: ```suggestion
### Microsoft Entra Token Credential (Recommended)
```

## Best Practices

> 1. **Use Entra ID TokenCredential** — Prefer AAD over connection strings/keys. Use `DefaultAzureCredential`only in development and specific credential in production.

Reviewer: ```suggestion
1. **Use Microsoft Entra Token Credential** — Prefer Microsoft Entra ID over connection strings/keys. Use `DefaultAzureCredential` only in development and specific credential in production.
```

|---------|---------|---------|
| `StorageSharedKeyCredential` | ✅ | ❌ |
| SAS generation | ✅ | ❌ |

> | Entra ID TokenCredential | ✅ | ❌ |

Reviewer: ```suggestion
| Microsoft Entra Token Credential | ✅ | ❌ |
```
